### PR TITLE
CON-2155 - Remove the legal entity from existing data

### PR DESCRIPTION
**Pull request by VasanthaKasirajan3008**

### CODE REVIEW

File: src/Data/Recruit.ChangeScripts/020-vacancyReviews_Unset_LegalEntityId.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        maxLoops = 1;
+    }
+    
+    do {
+        let matchedDocs = coll.aggregate([
+            {
+                $match: query
+            },
+            {
+                $sort: { "createdData": 1 }
```
Comment (cofaulco) [inline src/Data/Recruit.ChangeScripts/020-vacancyReviews_Unset_LegalEntityId.js:34]: is this meant to be "createdDate"

File: src/Data/Recruit.ChangeScripts/020-vacancyReviews_Unset_LegalEntityId.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        ]);
+
+        print(`Found ${matchedDocs._batch.length} document(s) to operate on in pass-through ${passThrough} of ${maxLoops}.`);
+
+        while (matchedDocs.hasNext()) {
+            let doc = matchedDocs.next();
+            
+            var updateDocument = {
+                $unset: {
+                    "vacancySnapshot.legalEntityId": 1
```
Comment (cofaulco) [inline src/Data/Recruit.ChangeScripts/020-vacancyReviews_Unset_LegalEntityId.js:51]: as per Mongodb unset documentation should this be:

$unset: {
    "vacancySnapshot.legalEntityId": ""
}

File: src/Data/Recruit.ChangeScripts/030-vacancies_Unset_LegalEntityId.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        maxLoops = 1;
+    }
+    
+    do {
+        let matchedDocs = coll.aggregate([
+            {
+                $match: query
+            },
+            {
+                $sort: { "createdData": 1 }
```
Comment (cofaulco) [inline src/Data/Recruit.ChangeScripts/030-vacancies_Unset_LegalEntityId.js:34]: "createdDate"

File: src/Data/Recruit.ChangeScripts/030-vacancies_Unset_LegalEntityId.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        ]);
+
+        print(`Found ${matchedDocs._batch.length} document(s) to operate on in pass-through ${passThrough} of ${maxLoops}.`);
+
+        while (matchedDocs.hasNext()) {
+            let doc = matchedDocs.next();
+            
+            var updateDocument = {
+                $unset: {
+                    "legalEntityId": 1
```
Comment (cofaulco) [inline src/Data/Recruit.ChangeScripts/030-vacancies_Unset_LegalEntityId.js:51]: same as above regarding unset

File: src/Data/Recruit.ChangeScripts/020-vacancyReviews_Unset_LegalEntityId.js (hunk 1 above)
Comment (VasanthaKasirajan3008) [reply src/Data/Recruit.ChangeScripts/020-vacancyReviews_Unset_LegalEntityId.js]: done

File: src/Data/Recruit.ChangeScripts/020-vacancyReviews_Unset_LegalEntityId.js (hunk 2 above)
Comment (VasanthaKasirajan3008) [reply src/Data/Recruit.ChangeScripts/020-vacancyReviews_Unset_LegalEntityId.js]: $unset: {"vacancySnapshot.legalEntityId": ""} and $unset: {"vacancySnapshot.legalEntityId": 1} returns the same result. I have changed  to $unset: {"vacancySnapshot.legalEntityId": ""}

File: src/Data/Recruit.ChangeScripts/030-vacancies_Unset_LegalEntityId.js (hunk 3 above)
Comment (VasanthaKasirajan3008) [reply src/Data/Recruit.ChangeScripts/030-vacancies_Unset_LegalEntityId.js]: done

**Review by cofaulco — APPROVED** [review]

**Pull request merged**